### PR TITLE
Notifications improvements

### PR DESCRIPTION
**Pull request by imorland**
- moved views from `/assets` to `/resources`
- Fix bug with fetching username from blueprint
- Refactor and use queues for notification sending
- updated alert template for new pd
- add notifications for new post in a pd

Screenshot 2020-02-14 at 19 52 33

Addresses #75

### CODE REVIEW

File: src/Listeners/QueueNotificationJobs.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            new Jobs\SendNotificationWhenPrivateDiscussionStarted($event->discussion, $event->newUsers)
+        );
+    }
+
+    public function postMadeInPrivateDiscussion(Saving $event)
+    {
+        $actor = $event->actor;
+
+        $event->post->afterSave(function ($post) use ($actor) {
+            if ($post->discussion->is_private && $post->number !== 1) {
```
Comment (datitisev) [inline src/Listeners/QueueNotificationJobs.php]: Not sure if I like this private check. What about other extensions that use private discussions (none right now, but imagine)?

I assume there's a way to check that the discussion is a byobu discussion, I think that'd be a better idea.

Comment (imorland) [reply src/Listeners/QueueNotificationJobs.php]: I think, given that `is_private` is not a part of byobu specifically, we should consider some sort of token to be included on the `discussions` table, to indicate which extension 'owns' it.

Probably something for a later date?

Comment (datitisev) [reply src/Listeners/QueueNotificationJobs.php]: What about checking for recipients? Notify if there are any recipients & the post isn't the discussion's first post.

Comment (imorland) [reply src/Listeners/QueueNotificationJobs.php]: Hmm, let me play around with that idea. `SendNotificationWhenPostedInPrivateDiscussion` does look there, but it makes sense to check here too.

Comment (imorland) [reply src/Listeners/QueueNotificationJobs.php]: Ok, here's one solution...

**Comment by datitisev** [comment]
Won't that always return truthy, as it returns a collection of users (even if it's empty)? Perhaps make it ->recipientUsers()->count() (I think that's a method on eloquent relationship... Not sure) or recipientIsers->length

**Comment by imorland** [comment]
recipientUsers is only present on a byobu discussion, so my understanding is that it will only return true if it’s a byobu private discussion. 

It certainly appears to work from some local testing.

**Comment by imorland** [comment]
I’ll step over it with the debugger attached tomorrow to double check

**Comment by imorland** [comment]
@datitisev just had a chance to check this out, you were correct, i've updated accordingly :)

**Pull request merged**